### PR TITLE
python and pack update

### PR DESCRIPTION
**Pull request by DegnKristine**
fixes: 
#170 - specifying the version in setup.py
#176 - Biopython > 1.78 - protein3to1. 
#183 - update of all packs 
#197 - fonts changed in newer version of matplotlib

### CODE REVIEW

**Comment by DegnKristine** [comment]
Note that we can omit the changes to setup.py since it is now up to date.

File: mutatex/utils.py [hunk 1]
```diff
@@ -86,21 +87,31 @@ def init_arguments(arguments, parser):
...
+    names = []
+    for fname in flist:
+        try:
+            name = matplotlib.font_manager.FontProperties(fname=fname).get_name()
+            names.append(name)
+        except:
+            pass
+    if type(names) is not str:
         return names
     return textwrap.fill(", ".join(sorted(list(set(names)))), width=69)
```
Comment (mtiberti) [inline mutatex/utils.py:104]: here it looks like we're not using the `return_as_string` argument as we were before the argument name change. It should probably be something like:

```
 def get_font_list(return_as_string=True):
    ....
    if not return_as_string:
        return names
    return ...
```

File: mutatex/utils.py [hunk 2]
```diff
@@ -356,7 +367,8 @@ def get_residue_list(infile, multimers=True, get_structure=False):
         sequences[chain_name] = ''
         for residue in chain:
             try:
-                res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
+                #res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
```
Comment (mtiberti) [inline mutatex/utils.py:370]: this comment is redundant since we use GitHub - can be removed

File: mutatex/utils.py [hunk 3]
```diff
@@ -378,7 +390,8 @@ def get_residue_list(infile, multimers=True, get_structure=False):
             for residue in model[cg[0]]:
                 resid = residue.get_id()[1]
                 try:
-                    res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
+                    #res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
```
Comment (mtiberti) [inline mutatex/utils.py:393]: same as per previous comment

File: mutatex/utils.py [hunk 4]
```diff
@@ -442,7 +455,8 @@ def get_foldx_sequence(pdb, multimers=True):
         chain_name = chain.get_id()
         for residue in chain:
             try:
-                res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
+                #res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
```
Comment (mtiberti) [inline mutatex/utils.py:458]: and this one as well

File: setup.py [hunk 5]
```diff
@@ -35,14 +35,14 @@
...
 
-    install_requires=[ 'biopython',
-                       'matplotlib',
-                       'numpy',
-                       'pandas',
-                       'logomaker',
-                       'scipy',
-                       'six',
-                       'openpyxl',
+    install_requires=[ 'biopython==1.84',
```
Comment (mtiberti) [inline setup.py:38]: why `==` and not `>=`? Feels like we are pinning a specific version for no real reason

File: setup.py [hunk 6]
```diff
@@ -35,14 +35,14 @@
...
+                       'matplotlib>=3.9',
+                       'numpy>=2.1.3',
+                       'pandas>=2.2.3',
+                       'logomaker==0.8',
```
Comment (mtiberti) [inline setup.py:42]: same here re pinning a specific version

File: setup.py [hunk 7]
```diff
@@ -35,14 +35,14 @@
...
+                       'scipy>=1.14.1',
+                       'six>=1.14.1',
+                       'openpyxl>=3.1.5',
```
Comment (mtiberti) [inline setup.py:45]: what is the reasoning behind these version specification? Just for me to know where these come from

**Review by mtiberti — CHANGES_REQUESTED** [review]

File: setup.py (hunk 7 above)
Comment (DegnKristine) [reply setup.py]: Reason for these versions: 

I Installed mutatex in a new python 3.12 environment without versioning the the packs. Once fixing the update and the code, I checked which versions of the packs were used and wrote these here. These are all the newest stable version of the packs. 

Essentially we don't gain anything from having the versions listed at the moment. Only when updates happen such as the biopython or matplotlib changes affects the utility of MutateX does it make sense to list specific versions. 

Suggestion - we drop these requirements, so the user simply download the newest versions, and instead leave a notice in the manual or .md that per 9th of December 2024 MutateX was updated to these versions of the packs, so the user can troubleshoot themself, open an issue but know the version to revert to. We can do so again in the future if we do a similar update.

Comment (DegnKristine) [reply setup.py]: Updated as per agreement.

**Review by mtiberti — APPROVED** [review]

**Pull request merged**